### PR TITLE
separating workers

### PR DESCRIPTION
**Pull request by nojaf**
See #67.
Not sure if I made things better.
The `test.fsx` represents code I have in my current service worker.
It is an improvement because of the latest ts2fable generation of [lib.webworker.d.ts](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.webworker.d.ts).

However I believe some manually work to Fable.Import.WebWorkers will still be required.

### CODE REVIEW

File: Browser/test.fsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    let json = 
+        ev.data
+        |> Option.map (fun data -> data.json())
+
+    match json with
+    | Some payload ->
+        let myInfo = payload :?> MyPushNotificationInfo
+        
+        let title = "My title"
+        let options:NotificationOptions =
```
Comment (nojaf) [inline Browser/test.fsx]: How should this be tackled? What should NotificationOptions be in this case? Just a plain record?

Comment (MangelMaxime) [reply Browser/test.fsx]: When facing this kind of situation I like to provide an helper on top of it.

Something like:

```
Notification.empty
|> Notification.body "my body"
|> Notification.icon "my-icon"
....
```

Like that the user only specify what is needed for him. This is the same approach I used in [`Thoth.Elmish.Toast`](https://mangelmaxime.github.io/Thoth/elmish/toast_demo.html) for example.

File: Browser/Fable.Helpers.WebWorker.fs [hunk 2]
```diff
@@ -0,0 +1,61 @@
+module Fable.Helpers.WebWorker
```
Comment (MangelMaxime) [inline Browser/Fable.Helpers.WebWorker.fs]: This files should be a standalone library.

`Fable.Import.Browser` package is only a binding and do not include the sources files. So the helpers can't be compiled.

I think having `Fable.Import.Browser` with both:
- Browser API
- WebWorker API

is fine. If people want a better UX then they will be able to add `Fable.Helpers.WebWorker` library.

**Comment by nojaf** [comment]
Might re-do this in the future sometime, no intentions to finish this in its current form.

**Pull request closed without merging**